packages/components/src/components/data-table/table-row-info.jsx

```diff
@@ -3,47 +3,35 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import ThemedScrollbars from '../themed-scrollbars';
 
-export default class TableRowInfo extends React.Component {
-    constructor(props) {
-        super(props);
+const TableRowInfo = ({ replace, is_footer, cells, className }) => {
+    const [show_details, setShowDetails] = React.useState(false);
 
-        this.state = {
-            showDetails: false,
-        };
-    }
-
-    toggleDetails = () => {
-        if (this.props.replace) {
-            this.setState(state => ({ showDetails: !state.showDetails }));
+    const toggleDetails = () => {
+        if (replace) {
+            setShowDetails(!show_details);
         }
     };
 
-    render() {
-        return (
-            <div
-                onClick={this.props.is_footer || !this.props.replace ? undefined : this.toggleDetails}
-                className={classNames(this.props.className, { 'statement__row--detail': this.state.showDetails })}
-            >
-                {this.state.showDetails ? (
-                    <ThemedScrollbars height='80px'>
-                        <div>
-                            {this.props.replace.component ? (
-                                this.props.replace.component
-                            ) : (
-                                <p className='statement__row--detail-text'>{this.props.replace.message}</p>
-                            )}
-                        </div>
-                    </ThemedScrollbars>
-                ) : (
-                    this.props.cells
-                )}
-            </div>
-        );
-    }
-}
+    return (
+        <div
+            onClick={is_footer || !replace ? undefined : toggleDetails()}
+            className={classNames(className, { 'statement__row--detail': show_details })}
+        >
+            {show_details ? (
+                <ThemedScrollbars height='80px'>
+                    <div>{replace?.component ?? <p className='statement__row--detail-text'>{replace.message}</p>}</div>
+                </ThemedScrollbars>
+            ) : (
+                cells
+            )}
+        </div>
+    );
+};
 
 TableRowInfo.propTypes = {
     cells: PropTypes.arrayOf(PropTypes.node),
     className: PropTypes.string,
     replace: PropTypes.object,
```

Comment:
```suggestion
    cells: PropTypes.arrayOf(PropTypes.node),
    className: PropTypes.string,
    replace: PropTypes.object,
    is_footer: PropTypes.bool,
```

Reply:
Is it possible to define the shape of the object for replace prop? I see we are relying on that to have `message` and `component` properties itself. It is good to have this upfront so the user of the component knows about it.